### ISSUE TITLE
Could not establish Memcached connection.

### ISSUE THREAD

**Issue by maskwang** [b3it/laravel-memcached-plus#1]
```
 if (in_array('255.255.255', $memcachedStatus) && count(array_unique($memcachedStatus)) === 1) {
            throw new RuntimeException("Could not establish Memcached connection.");
        }
```

hi~ I'm from China. If I add one server, then will throw this error!  
actually memcached has established.

**Comment by tomcastleman** [comment]
Hi @maskwang - sorry I didn't get notified of this issue I only just saw it.

I think you misunderstand this check - it is correct (it has been reviewed by @taylorotwell and merged into laravel/framework too). See [this commit](https://github.com/laravel/framework/commit/618cb5a665e10361f711fa04588dfd5ca2647d45) which is still present in the latest branch [here](https://github.com/laravel/framework/blob/master/src/Illuminate/Cache/MemcachedConnector.php#L37).

Please note:
- [`Memcached::getVersion()`](http://php.net/manual/en/memcached.getversion.php) returns an array of added servers with their versions:

```
Array
(
    [localhost:11211] => 1.2.6
)
```
- Where a connection has failed to a server the version is returned as '255.255.255'.
- This check throws a `RuntimeException` when `255.255.255' is present in the array, _and_ there is only one server in the array.

If this check is throwing an exception for you with one server, that must mean the connection to that server has failed.

Apologies again for the delay in my reply.

Thanks

**Comment by tomcastleman** [comment]
@maskwang Please remove your fork from Packagist. You have implemented your (incorrect) 'fix', changed the namespace, and submitted it as a new package. If you insist on using your version of this package, you can do so by following [these instructions](https://getcomposer.org/doc/05-repositories.md#vcs).

[Packagist](https://packagist.org/packages/submit) clearly states when submitting a new package *Do not submit forks of existing packages*, yet that is exactly what you have done. This is not really acceptable - please remove it.

I am all for collaboration and contributions, please continue to do so via Github issue and PRs.

Thanks

**Comment by totoleo** [comment]
I also meet the problem. But it`s confused me that I  get the cached data from server after meeting the error.

**Comment by gilbertoalbino** [comment]
The Exceptions says:

> Could not establish Memcached connection

So, PHP Memcached "extension" can not "connect" to "Memcached".

If you are on Linux, please, try installing:

```
sudo apt-get install memcached
sudo service memcached status
```
And you are ready to go!

**Comment by khsing** [comment]
I actually meet this problem also. But the problem is I am using a Cloud based memcache protocol compatible service via Aliyun. Which can't return the right version of memcached, but the memcache function is normal. 

```php
$memcache = new Memcached();
$memcache->addServer($host,$port);
$memcache->get('somekey');
```
All above works fine.

but  `Cache::get('somekey')` will fail and the error will be

```
RuntimeException with message 'Could not establish Memcached connection.'
```

If possible I think add a option to turn on/off the version check might be a better solution.

**Comment by tomcastleman** [comment]
Possibly related [laravel/framework issue](https://github.com/laravel/framework/issues/17957)

**Comment by tomcastleman** [comment]
Fixed in #10
